apps/api/src/sandbox/services/snapshot.service.ts

```diff
@@ -105,30 +113,47 @@ export class SnapshotService {
     ref: string,
     entrypoint: string[] | undefined,
     skipValidation: boolean,
+    regionId: string,
   ): Promise<Snapshot | null> {
-    // Check if there is already an active snapshot with the same ref;
+    // Check if there is already an active snapshot with the same ref in the provided region
     // Only check entrypoint if skipValidation is not set on the DTO
     // We can skip the pulling and validation in that case - note: relevant only for Docker
 
-    const snapshotFindOptions: any = {
-      ref,
-      state: SnapshotState.ACTIVE,
-    }
-
     if (!entrypoint || entrypoint.length === 0) {
       return null
     }
 
+    const qb = this.snapshotRepository
```

Comment:
I suggest we remove `snapshot` from this join chain.

If a snapshot runner exists, the snapshot is active implicitly.

The method can then just return a boolean instead of the full entity. 

(The query can be an exists query as well)